src/scripts/createLocationAutocompleteData/index.ts

```diff
@@ -0,0 +1,345 @@
+    locationDataRows.reduce((relevantLocations, location, rowIndex) => {
+      logProgress(rowIndex, locationRowCount);
+
+      const [
+        geonameId,
+        name,
+        _asciiname,
+        _alternatenames,
+        _latitude,
+        _longitude,
+        featureClass,
+        featureCode,
+        _countryCode,
+        _cc2,
+        admin1Code,
+        _admin2Code,
+        _admin3Code,
+        _admin4Code,
+        population,
+        _elevation,
+        _dem,
+        _timezone,
+        _modificationDate,
+      ] = location.split("\t") as TLocationData; // lines are tab delimited
+
+      const isPopulatedPlaceOfInterest =
+        featureClass === allowedFeatureClass &&
+        allowedFeatureCodes.includes(featureCode);
+      const hasPopulation = Number(population) !== 0;
+
+      if (isPopulatedPlaceOfInterest && hasPopulation) {
+        const alternateNames = parsedAlternateNames.filter(
+          ({ alternateOf, name: alternateName }) =>
+            alternateOf === geonameId && alternateName !== name
+        );
+        const preferredName = alternateNames.find(
+          ({ isPreferredName }) => isPreferredName === "1"
+        );
+        const alternateNamesFinal = alternateNames.map((alternateName) => {
+          // Include the original name as an alternative name if we’ll use an
+          // alternate name that is the preferred name.
+          if (preferredName && preferredName.name === alternateName.name) {
+            return name;
+          }
+
+          return alternateName.name;
+        });
+
+        relevantLocations.push({
+          id: geonameId,
+          // switch names if an alternate name is the preferred location name
+          name: preferredName ? preferredName.name : name,
+          stateCode: admin1Code,
+          countryCode: "USA",
+          featureClass,
+          featureCode,
+          population,
+          ...(alternateNames &&
+            alternateNames.length > 0 && {
+              alternateNames: alternateNamesFinal,
+            }),
+        });
+      }
+
+      return relevantLocations;
+    }, Array());
+
+  // Filter out locations that have another populated place as a parent.
+  console.info("Filtering by hierachy");
+  const locationDataPopulatedCount = locationDataPopulated.length;
+  const locationDataPopulatedTopLevel = locationDataPopulated.filter(
+    (locationPopulated, rowIndex) => {
+      logProgress(rowIndex, locationDataPopulatedCount);
+
+      const hasPopulatedParentLocation = hierarchyIds.some(
+        ([parentId, childId]) => {
+          if (locationPopulated.id === childId) {
+            return locationDataPopulated.some((location) => {
+              if (
+                location.id === parentId &&
+                location.featureClass === allowedFeatureClass
+              ) {
+                return true;
+              }
+              return false;
+            });
+          }
+          return false;
+        }
```

Comment:
Not necessary at all, but you can shave off some indenting if you put the bailout conditions first (no guarantee that I didn't make mistakes below, typing in the GitHub text box):

```suggestion
      const hasPopulatedParentLocation = hierarchyIds.some(
        ([parentId, childId]) => {
          if (locationPopulated.id !== childId) {
            return false;
          }
          return locationDataPopulated.some((location) => {
            return (
              location.id === parentId &&
              location.featureClass === allowedFeatureClass
            );
          });
          
        }
```

Reply:
That does look better: [9d94401](https://github.com/mozilla/blurts-server/pull/3186/commits/9d94401ff628462e4795246d0dc64dc14d86931e).